tests/ui/resolve/issue-112713.rs

```diff
@@ -0,0 +1,16 @@
+// edition: 2021
+
+pub fn foo() -> u32 {
+    use sub::*;
+    C //~ERROR `C` is ambiguous
```

Comment:
This is a breaking change: This is a breaking change. It will throw an ambiguous error after this PR, but it currently compiles without error.